webhook_router/app.py

```diff
@@ -21,38 +98,108 @@ def handle_github_webhook() -> tuple[str, int]:
 
     Returns:
         A tuple containing an empty string and 200 status code on success or
-        a failure message and 403 status code.
+        a failure message and 4xx status code.
     """
     if secret := app.config.get("WEBHOOK_SECRET"):
-        if not (signature := request.headers.get(WEBHOOK_SIGNATURE_HEADER)):
+        is_valid, error = _validate_signature(secret)
```

Comment:
I really like this type of error checking from the new languages like Go/Rust, not sure if it is Pythonic or others agree, just leaving a comment :D

Reply:
Tbh, I am not a big fan of it and would rather raise an exception and am not sure if its pythonic, but I followed @jdkandersson's advice from https://github.com/canonical/github-runner-webhook-router/pull/3#discussion_r1524245306 (see commit https://github.com/canonical/github-runner-webhook-router/pull/3/commits/a3c3c1f2f762358054bcfd6bc1884159e7cfcd9f) here.